lms/djangoapps/teams/csv.py

```diff
@@ -406,21 +432,31 @@ def is_FERPA_bubble_breached(self, teamset_id, team_name):
         else:
             return True
 
-    def validate_proposed_team_size_wont_exceed_maximum(self, team_name, teamset_id):
+    def validate_team_sizes_not_exceeded(self):
         """
         Validates that the number of users we want to add to a team won't exceed maximum team size.
         """
-        if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
-            max_team_size = self.course.teams_configuration.default_max_team_size
-        else:
-            max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+        for teamset_id in self.teamset_ids:
+            # Get max size for team-set
+            if self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size is None:
+                max_team_size = self.course.teams_configuration.default_max_team_size
+            else:
+                max_team_size = self.course.teams_configuration.teamsets_by_id[teamset_id].max_team_size
+
+            if max_team_size is not None:
+                # Get teams in teamset
+                team_names = [
+                    teamset_to_team[1] for teamset_to_team in self.user_count_by_team
+                    if teamset_to_team[0] == teamset_id
+                ]
+
+                for team_name in team_names:
+                    # Calculate team size
+                    key = (teamset_id, team_name)
+                    if (self.user_count_by_team[key] - self.user_to_remove_by_team[key]) > max_team_size:
```

Comment:
would it be possible for us to combine these two dicts and just have something like self.proposed_team_sizes where we both add and subtract team members?

Reply:
Yes, was trying to see if this was possible w/in the current architecture but I agree it would be cleaner to do it that way. Will play around with refactoring that.

Reply:
Removed `self.user_to_remove_by_team` and keep a running count in `user_count_by_team` in [this commit](https://github.com/edx/edx-platform/pull/24290/commits/966aedafd674896357c8a0c3b5cd3c6090d9cad0)